src/app/features/tech-record/components/tech-promote/tech-promote.component.ts

```diff
@@ -1,58 +1,80 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { Validators } from '@angular/forms';
 import { ActivatedRoute, Router } from '@angular/router';
 import { GlobalErrorService } from '@core/components/global-error/global-error.service';
 import { CustomFormControl, CustomFormGroup, FormNodeTypes } from '@forms/services/dynamic-form.types';
 import { StatusCodes, TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { Actions, ofType } from '@ngrx/effects';
 import { RouterReducerState } from '@ngrx/router-store';
 import { select, Store } from '@ngrx/store';
 import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
 import { selectRouteNestedParams } from '@store/router/selectors/router.selectors';
-import { updateEditingTechRecord, updateTechRecords } from '@store/technical-records';
+import {
+  archiveTechRecord,
+  archiveTechRecordSuccess,
+  updateEditingTechRecord,
+  updateTechRecords,
+  updateTechRecordsSuccess
+} from '@store/technical-records';
 import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
-import cloneDeep from 'lodash.clonedeep';
 import { Observable, take, tap } from 'rxjs';
 
 @Component({
   selector: 'app-tech-promote',
   templateUrl: './tech-promote.component.html'
 })
-export class TechPromoteComponent {
+export class TechPromoteComponent implements OnInit {
   vehicleTechRecord$: Observable<VehicleTechRecordModel | undefined>;
 
   techRecord?: TechRecordModel;
 
   form: CustomFormGroup;
 
+  buttonLabel: string;
+
   constructor(
+    private actions$: Actions,
     private errorService: GlobalErrorService,
     private route: ActivatedRoute,
     private router: Router,
     private routerStore: Store<RouterReducerState>,
     private techRecordsStore: Store<TechnicalRecordServiceState>,
     private technicalRecordService: TechnicalRecordService
   ) {
-    this.errorService.clearErrors();
-
     this.vehicleTechRecord$ = this.technicalRecordService.selectedVehicleTechRecord$.pipe(
-      tap(
-        vehicleTechRecord =>
-          (this.techRecord = cloneDeep(vehicleTechRecord?.techRecord.find(record => record.statusCode === StatusCodes.PROVISIONAL))!)
-      )
+      tap(vehicle => this.technicalRecordService.viewableTechRecord$(vehicle!).subscribe(techRecord => (this.techRecord = techRecord)))
```

Comment:
A subscribe inside a tap feel counter intuitive. Would it not be nicer to pipe the emitted value of the outer observable into a new inner one? try using `switchMap` like `...pipe(<take>, switchMap( v => this.technicalRecordService.viewableTechRecord$(v!))).subscribe...`

Reply:
Addressing your comment above resolved this issue as well.